dao/src/main/java/org/thingsboard/server/dao/sql/resource/TbResourceInfoRepository.java

```diff
@@ -46,4 +46,29 @@ Page<TbResourceInfoEntity> findAllTenantResourcesByTenantId(@Param("tenantId") U
     Page<TbResourceInfoEntity> findTenantResourcesByTenantId(@Param("tenantId") UUID tenantId,
                                                              @Param("searchText") String searchText,
                                                              Pageable pageable);
+
+    @Query("SELECT tr FROM TbResourceInfoEntity tr WHERE " +
```

Comment:
Plz add `:resourceType IS NULL OR tr.resourceType = :resourceType"` to the existing queries.